### ISSUE TITLE
question about data load & hue consistency

### ISSUE THREAD

**Issue by D222097** [yayoyo66/RDIStitcher#2]
Hi, I wanted to check if my data has been loaded correctly into the project. Could someone help verify that everything is in order?
input data:
| mask1 | mask2| input img|
|--------|--------|-----------|
| ![001](https://github.com/user-attachments/assets/c07566b1-7b80-438c-9316-66503e651b44) |![002](https://github.com/user-attachments/assets/37ad877a-81fb-4010-b004-d084d1a21e05)|![001](https://github.com/user-attachments/assets/3e1d0902-45f4-497f-b895-41a206026245)|

and get processed data:
| images |conditioning_images| masks | weightings |
|--------|--------|-----------|-----------|
|![a-images](https://github.com/user-attachments/assets/2ca94b11-ba69-4143-86ca-2cd075aa7201)|![a-conditioning_images](https://github.com/user-attachments/assets/77db4109-099e-46f0-b943-96c19e6ba380)|![a-masks](https://github.com/user-attachments/assets/e648130c-712a-4fbf-9f36-036959816715)|![a-weightings](https://github.com/user-attachments/assets/76b1405b-ed9c-49dc-9c57-f730a1926995)|

Additionally, there is another question: It seems that both the input image and the condition image add color jitter in the same areas. How does the model learn to achieve hue consistency? Alternatively, how can we ensure that the model is not learning to inpaint the missing areas based on the right image itself, but rather using the left image to fill in the missing areas while maintaining hue consistency?

**Comment by yayoyo66** [comment]
Yeah, your input data and processed data are correct. 

**Q1:  It seems that both the input image and the condition image add color jitter in the same areas.**

My model is to **inpaint the missing part of the right image based on the left image**, that is, the white area in the _masks_ is the area that needs to be inpainted. 

Therefore, the left image is consistent in _images_ and _conditioning_images_, and I do not require the model to modify this part.

**Q2:  Alternatively, how can we ensure that the model is not learning to inpaint the missing areas based on the right image itself, but rather using the left image to fill in the missing areas.**

In this sense, I was inspired by the work of [LeftRefill](https://github.com/ewrfcas/LeftRefill), which showed that this type of input allows the model to repaint the right based on the left. This is interesting finding. 

**Q3:  How does the model learn to achieve hue consistency?**

In fact, the large-scale pre-trained inpainting model (our base model) itself has the ability to provide hue consistency. I mean , keeping the inpainted regions in the same hue as the original image is a fundamental ability of an inpainting model, so we can use this ability in image stitching tasks. 

According to my experimental results, the trained model can still have a certain degree of hue consistency even without adding colour jitter to augment the training images. 

Adding colour jitter leads to higher success rates and better results (this is what I mean by improved robustness in the paper), which is an intuitive and easy to interpret result.  I do not perform ablation experiments to verify this, because it is too difficult to find an appropriate metric.
